coriolis/tests/osmorphing/netpreserver/test_nmconnection.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2025 Cloudbase Solutions Srl
+# All Rights Reserved.
+
+from unittest import mock
+
+from coriolis.osmorphing import base
+from coriolis.osmorphing.netpreserver import nmconnection
+from coriolis.osmorphing import redhat
+from coriolis.tests import test_base
+
+
+class NmconnectionNetPreserverTestCase(test_base.CoriolisBaseTestCase):
+    """Test suite for the NmconnectionNetPreserver class."""
+
+    def setUp(self):
+        super(NmconnectionNetPreserverTestCase, self).setUp()
+        self.event_manager = mock.MagicMock()
+        self.detected_os_info = {
+            'os_type': 'linux',
+            'distribution_name': redhat.RED_HAT_DISTRO_IDENTIFIER,
+            'release_version': '6',
+            'friendly_release_name': mock.sentinel.friendly_release_name,
+        }
+        self.netpreserver = nmconnection.NmconnectionNetPreserver(
+            redhat.BaseRedHatMorphingTools(
+                mock.sentinel.conn, mock.sentinel.os_root_dir,
+                mock.sentinel.os_root_dir, mock.sentinel.hypervisor,
+                self.event_manager, self.detected_os_info,
+                mock.sentinel.osmorphing_parameters,
+                mock.sentinel.operation_timeout)
+        )
+
+    @mock.patch.object(base.BaseLinuxOSMorphingTools, '_list_dir')
+    def test__get_nmconnection_files(self, mock_list_dir):
+        mock_list_dir.return_value = [
+            'eth0.nmconnection', 'eth1.nmconnection', 'other-file']
+        result = self.netpreserver._get_nmconnection_files(
+            self.netpreserver.nmconnection_file)
+        expected_result = [
+            'etc/NetworkManager/system-connections/eth0.nmconnection',
+            'etc/NetworkManager/system-connections/eth1.nmconnection'
+        ]
+        mock_list_dir.assert_called_once_with(
+            self.netpreserver.nmconnection_file)
+        self.assertEqual(result, expected_result)
+
+    @mock.patch.object(base.BaseLinuxOSMorphingTools, '_read_config_file')
+    @mock.patch.object(nmconnection.NmconnectionNetPreserver,
+                       '_get_nmconnection_files')
+    def test__get_keyfiles_by_type(self, mock_get_nmconnection_files,
+                                   mock_read_config_file):
+        mock_get_nmconnection_files.return_value = [mock.sentinel.nmconn_file]
+        mock_read_config_file.side_effect = [{"type": "ethernet"}]
+
+        result = self.netpreserver._get_keyfiles_by_type(
+            "ethernet", self.netpreserver.nmconnection_file)
+
+        mock_get_nmconnection_files.assert_called_once_with(
+            self.netpreserver.nmconnection_file)
+        mock_read_config_file.assert_called_once_with(
+            mock.sentinel.nmconn_file)
+        self.assertEqual(result, [(mock.sentinel.nmconn_file,
+                                   {"type": "ethernet"})])
+
+    @mock.patch.object(base.BaseLinuxOSMorphingTools, '_test_path')
+    @mock.patch.object(base.BaseLinuxOSMorphingTools, '_list_dir')
```

Comment:
Same as ifcfgs. Please only mock tested method calls, and not inner calls from down the stack.